coderd/tailnet.go

```diff
@@ -455,3 +492,18 @@ func (s *ServerTailnet) Close() error {
 	<-s.derpMapUpdaterClosed
 	return nil
 }
+
+type instrumentedConn struct {
+	net.Conn
+
+	agentID       uuid.UUID
+	closeOnce     sync.Once
+	connsPerAgent *prometheus.GaugeVec
+}
+
+func (c *instrumentedConn) Close() error {
+	c.closeOnce.Do(func() {
```

Comment:
are network connections _always_ explicitly closed?

Reply:
This one I'm not 100% sure on. Admittedly, I found this idea from a stackoverflow post which seemed to work for a couple other people. Was planning to get this into dev and monitor to make sure it works as intended with a lot more usage than I can reproduce myself.

Reply:
I guess maybe they _should be_ and this gauge will tell us if there's a leak...